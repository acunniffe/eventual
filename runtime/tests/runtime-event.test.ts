/// Example generated code

import EventualEventType from '../src/index';

export interface IAUserWithEmailAndIdSignedUp {
  email: string;
  id: string;
}
export const AUserWithEmailAndIdSignedUp = new EventualEventType<
  IAUserWithEmailAndIdSignedUp
>(
  'AUserWithEmailAndIdSignedUp',
  (props) => `A #user with ${props.email} and ${props.id} signed-up`,
  {
    type: 'object',
    required: ['email', 'id'],
    additionalProperties: false,
    properties: { email: { type: 'string' }, id: { type: 'string' } },
  },
  ['user'],
  'USR-1'
);

test('runtime events can be created with valid props', () => {
  const newInstance = AUserWithEmailAndIdSignedUp.new({
    email: 'a@gmail.com',
    id: 'abcdefg',
  });
  expect(newInstance.toSentence()).toMatchSnapshot();
  expect(newInstance.toJson()).toMatchSnapshot();
});

test('runtime events will throw is created with wrong data', () => {
  expect(() =>
    AUserWithEmailAndIdSignedUp.new({ world: 'hello' })
  ).toThrowErrorMatchingSnapshot();
});

test('can validate invalid data w/o throwing', () => {
  expect(AUserWithEmailAndIdSignedUp.validate({ world: 'hello' })).toBe(false);
});
