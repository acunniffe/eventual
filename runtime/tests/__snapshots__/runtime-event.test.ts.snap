// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtime events can be created with valid props 1`] = `"A #user with $email [1m[4ma@gmail.com[24m[22m and $id [1m[4mabcdefg[24m[22m signed-up"`;

exports[`runtime events can be created with valid props 2`] = `
Object {
  "data": Object {
    "email": "a@gmail.com",
    "id": "abcdefg",
  },
  "type": "AUserWithEmailAndIdSignedUp",
}
`;

exports[`runtime events will throw is created with wrong data 1`] = `"Invalid props for Event Type AUserWithEmailAndIdSignedUp"`;
